packages/modules/vehicles/mercedeseq/config.py

```diff
@@ -0,0 +1,42 @@
+from typing import Optional
+
+
+class MercedesEQSocToken:
+    def __init__(self,
+                 refresh_token: str = "",
+                 access_token: str = "",
+                 expires_in: int = 0,
+                 id_token: str = "",
+                 token_type: str = "") -> None:
+        self.refresh_token = refresh_token
+        self.access_token = access_token
+        self.expires_in = expires_in
+        self.id_token = id_token
+        self.token_type = token_type
+
+
+class MercedesEQSocConfiguration:
+    def __init__(self, client_id: Optional[str] = None, client_secret: Optional[str] = None, vin: Optional[str] = None,
+                 callbackurl: Optional[str] = None, token: MercedesEQSocToken = None) -> None:
+        self.client_id = client_id          # show in UI
+        self.client_secret = client_secret  # show in UI
+        self.vin = vin                      # show in UI
+        self.callbackurl = callbackurl      # show in UI
```

Comment:
Wenn ich das richtig sehe, ist die Callback URL doch von Deiner Implementierung vorgegeben. Eine Anpassung vom Benutzer macht daher keinen Sinn. Daher würde ich die URL aus der Konfiguration entfernen.

Reply:
Nein, die Callback URL muss entweder mit der IP Adresse oder dem Hostname der eigenen Installation konfiguriert werden. Als default könnte man natürlich sowas einsetzen wie 
http://openwb/openWB/packages/modules/vehicles/mercedeseq/callback_ev.php
aber es kommt eben drauf an, ob der Anwender den hostname der openWB verändert hat oder nicht.

Reply:
In meiner Test V2 wären die möglichen callback URLs z.B.

```
http://raspi3b/openWB/packages/modules/vehicles/mercedeseq/callback_ev.php
http://raspi3b.fritz.box/openWB/packages/modules/vehicles/mercedeseq/callback_ev.php
http://192.168.178.89/openWB/packages/modules/vehicles/mercedeseq/callback_ev.php
```

Reply:
Der Hostname wir automatisch im UI gesetzt. Es wird einfach `location.origin` genommen.

Reply:
Sieht dann aktuell so aus.
![image](https://user-images.githubusercontent.com/7837691/222142928-5888f424-a596-46c9-9e7d-be6f2ad7aeae.png)

Reply:
Weiterhin werde ich die `callback_ev.php` in das UI Repo verschieben. Dann ist es analog zu dem Tesla Modul und alles, was zum UI gehört zusammen. Da brauchst Du Dich aber nicht drum kümmern.

Reply:
Wir brauchen aber definitiv die Callback_url angezeigt, damit der Anwender diese dann auch im Mercedes Developer Portal eingeben kann. Das muss identisch sein, das wird überprüft, sonst fällt die Authentifizierung auf die Nase. Kannst du das machen?

Reply:
Wenn du die callback_ev.php verschiebst, liegt sie dann in einem anderen Pfad? Der Pfad der callback_ev.php ist ja in der URL zur Authentifizierung enthalten als redirect_uri
`https://ssoalpha.dvb.corpinter.net/v1/auth?response_type=code&state=19&client_id=b85c982a-c432-4ef5-8f59-2a34d1f066d8&redirect_uri=http://192.168.178.89/openWB/packages/modules/vehicles/mercedeseq/callback_ev.php&scope=mb:vehicle:mbdata:evstatus%20offline_access%20openid`

Reply:
> Wir brauchen aber definitiv die Callback_url angezeigt, damit der Anwender diese dann auch im Mercedes Developer Portal eingeben kann. Das muss identisch sein, das wird überprüft, sonst fällt die Authentifizierung auf die Nase. Kannst du das machen?

Klar.

Reply:
> Wenn du die callback_ev.php verschiebst, liegt sie dann in einem anderen Pfad? Der Pfad der callback_ev.php ist ja in der URL zur Authentifizierung enthalten als redirect_uri `https://ssoalpha.dvb.corpinter.net/v1/auth?response_type=code&state=19&client_id=b85c982a-**********66d8&redirect_uri=http://192.168.178.89/openWB/packages/modules/vehicles/mercedeseq/callback_ev.php&scope=mb:vehicle:mbdata:evstatus%20offline_access%20openid`

Ist schon berücksichtigt.

Reply:
```suggestion
```